https://bugs.gentoo.org/723468#c5
By Anon Emuss <abuslbea@centurylink.net>
----
Fix minor HTML changes in tests

Newer versions of sphinx now put <p> tags around various elements.
Update the test_lists and CustomExtensionTests tests to expect that.
Also do not expect a class="first" for the Contents in the
CustomExtensionTests.

Tested and appears to work with sphinx-2.4.4 and sphinx-3.2.1.

diff --git a/tests/test_sphinx.py b/tests/test_sphinx.py
index 06078b1..8f4acb2 100644
--- a/tests/test_sphinx.py
+++ b/tests/test_sphinx.py
@@ -142,17 +142,17 @@ class GenericTests(SphinxIntegrationTests):
         output = self.read_file('index.html')
         self.assertIn(
             ('<ul class="simple">\n'
-             '<li>Item A</li>\n'
-             '<li>Item B</li>\n'
-             '<li>Item C</li>\n'
+             '<li><p>Item A</p></li>\n'
+             '<li><p>Item B</p></li>\n'
+             '<li><p>Item C</p></li>\n'
              '</ul>'),
             output
         )
         self.assertIn(
             ('<ol class="simple">\n'
-             '<li>Item 1</li>\n'
-             '<li>Item 2</li>\n'
-             '<li>Item 3</li>\n'
+             '<li><p>Item 1</p></li>\n'
+             '<li><p>Item 2</p></li>\n'
+             '<li><p>Item 3</p></li>\n'
              '</ol>'),
             output
         )
@@ -207,16 +207,17 @@ class CustomExtensionTests(SphinxIntegrationTests):
     def test_integration(self):
         output = self.read_file('index.html')
         self.assertIn('<table ', output)
-        self.assertIn('<th class="head">abc</th>', output)
-        self.assertIn('<th class="head">data</th>', output)
+        self.assertIn('<th class="head"><p>abc</p></th>', output)
+        self.assertIn('<th class="head"><p>data</p></th>', output)
         self.assertIn('</table>', output)
 
         self.assertIn(
             ('<div class="contents topic" id="contents">\n'
-             '<p class="topic-title first">Contents</p>\n'
+             '<p class="topic-title">Contents</p>\n'
              '<ul class="simple">\n'
-             '<li><a class="reference internal" href="#header" id="id1">Header</a><ul>\n'
-             '<li><a class="reference internal" href="#header-2" id="id2">Header 2</a></li>\n'
+             '<li><p><a class="reference internal" href="#header" id="id1">Header</a></p>\n'
+             '<ul>\n'
+             '<li><p><a class="reference internal" href="#header-2" id="id2">Header 2</a></p></li>\n'
              '</ul>\n</li>\n</ul>'),
             output
             )
